File: extract_pdf.js
const { PDFParse } = require('pdf-parse');
const fs = require('fs');

const pdfPath = './Prism of Magic_ Homebrew System for D&D 5e - 11-17.pdf';

async function extractPages() {
  try {
    const dataBuffer = fs.readFileSync(pdfPath);
    const parser = new PDFParse({ data: dataBuffer });
    const data = await parser.getText({ first: 4, last: 14 });
    
    // Extract text from all pages
    const fullText = data.text;
    
    // Try to identify pages 4-14
    // PDF-parse doesn't give us page-by-page, so we'll get all text
    // and look for patterns that indicate spell lists
    console.log('=== PDF Text Extraction ===');
    console.log('Total pages:', data.numpages);
    console.log('\n=== Full Text (looking for spell-to-prism mappings) ===\n');
    console.log(fullText);
    
    // Also try to extract structured data
    const lines = fullText.split('\n');
    const spellMappings = {};
    let currentPrism = null;
    
    for (let i = 0; i < lines.length; i++) {
      const line = lines[i].trim();
      
      // Look for prism names (might be in all caps or title case)
      if (line.match(/^(ABJURATION|CONJURATION|DIVINATION|ENCHANTMENT|EVOCATION|ILLUSION|NECROMANCY|TRANSMUTATION|Prism|PRISM)/i)) {
        currentPrism = line;
        continue;
      }
      
      // Look for spell names (common pattern: spell name followed by level)
      // Spell names are usually title case and might be followed by numbers or other info
      if (line && currentPrism && line.match(/^[A-Z][a-z]+(\s+[A-Z][a-z]+)*/)) {
        // This might be a spell name
        const spellName = line.split(/\s+/)[0] + (line.split(/\s+/)[1] ? ' ' + line.split(/\s+/)[1] : '');
        if (spellName.length > 2 && spellName.length < 50) {
          spellMappings[spellName] = currentPrism;
        }
      }
    }
    
    console.log('\n=== Extracted Spell Mappings ===');
    console.log(JSON.stringify(spellMappings, null, 2));
    
  } catch (error) {
    console.error('Error extracting PDF:', error);
  }
}

extractPages();

File: extract_pdf.py
#!/usr/bin/env python3
"""Extract text from PDF pages 4-14 to understand spell-to-prism mappings."""

try:
    import PyPDF2
    HAS_PYPDF2 = True
except ImportError:
    HAS_PYPDF2 = False

try:
    import pdfplumber
    HAS_PDFPLUMBER = True
except ImportError:
    HAS_PDFPLUMBER = False

import sys

def extract_with_pypdf2(pdf_path):
    """Extract text using PyPDF2."""
    with open(pdf_path, 'rb') as file:
        pdf_reader = PyPDF2.PdfReader(file)
        text = ""
        # Pages 4-14 (0-indexed: 3-13)
        for page_num in range(3, min(14, len(pdf_reader.pages))):
            page = pdf_reader.pages[page_num]
            text += f"\n--- Page {page_num + 1} ---\n"
            text += page.extract_text()
        return text

def extract_with_pdfplumber(pdf_path):
    """Extract text using pdfplumber."""
    with pdfplumber.open(pdf_path) as pdf:
        text = ""
        # Pages 4-14 (0-indexed: 3-13)
        for page_num in range(3, min(14, len(pdf.pages))):
            page = pdf.pages[page_num]
            text += f"\n--- Page {page_num + 1} ---\n"
            text += page.extract_text()
        return text

if __name__ == "__main__":
    pdf_path = "Prism of Magic_ Homebrew System for D&D 5e - 11-17.pdf"
    
    if HAS_PDFPLUMBER:
        print(extract_with_pdfplumber(pdf_path))
    elif HAS_PYPDF2:
        print(extract_with_pypdf2(pdf_path))
    else:
        print("ERROR: No PDF library available. Please install pdfplumber or PyPDF2:")
        print("  pip install pdfplumber")
        print("  or")
        print("  pip install PyPDF2")
        sys.exit(1)

